tests/scanner/audit/retention_rules_engine_test.py

```diff
@@ -239,11 +239,17 @@ def test_invalid_rule_with_no_res_id(self):
     maximum_retention: 365
 
 """
+
+    def test_number_of_rules(self):
+        """The number of rules should be exactly the same as the length of SUPPORTED_RETENTION_RES_TYPES."""
+        rules_engine = get_rules_engine_with_rule(RetentionRulesEngineTest.yaml_str_only_max_retention)
+        self.assertTrue(len(rre.SUPPORTED_RETENTION_RES_TYPES) == len(rules_engine.rule_book.resource_rules_map))
```

Comment:
Use a literal constant here, and use `assertEqual`, e.g. `self.assertEqual(2, len(...))`

You should also check that the number of rules is correct, i.e. 1 rule for buckets, 0 rules for tables.